llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1660,6 +1661,11 @@ static void computeKnownBitsFromOperator(const Operator *I,
 
         // We need to take the minimum number of known bits
         KnownBits Known3(BitWidth);
+        if (BitWidth != getBitWidth(L->getType(), Q.DL)) {
```

Comment:
If the pointer width is different from the index width, the optimization will be disabled. Is there a real target satisfying the condition?

Reply:
Yes this is happening in a lot of workloads in practice, e.g. index width 64 bits and GEPs with i32 indices.

Reply:
I think it is ok to fall through as the result is guarded by `std::min(Idx.countMinTrailingZeros(), Ptr.countMinTrailingZeros())`.